Feature/173 garden mobile

simalguven opened the pull request:
Including task creation and other features of garden implementation

SuleymanTolgaAcar commented:
I think you forgot to add the "node_modules" folder in the .gitignore file. There are more than 1.5m lines of change now :)

simalguven commented:
## Feature Description
<!-- Provide a detailed description of the new feature -->
Added three major functional tabs:
- Gardens Tab: Displays all gardens and allows creation (if user has permission).
- My Tasks Tab: Lists tasks assigned to the user, grouped by gardens, with task detail and completion options.
- Community Tab: Displays other users who share gardens with the current user. Users can follow each other directly.

Replaced the default Home tab with a custom welcome screen including a visual banner.
Enabled garden creation, depending on manager/worker enabled task creation in gardens and assigning tasks, accepting and declining tasks. 
Task viewing in backend altered; accept_task, create_task etc. added.
Tasks in a garden are displayed depending on that users role in that garden. 
Under garden detail screen tasks are shown under pending/in progress/completed plus members of that garden and a calendar for that gardens tasks are displayed . 
Under community tab users are shown with all other users which are also members of the gardens current user is already a member of. From there a user can follow other users . 
In community tab or in my profile/followers-following users are clickable and our current user is forwarded to view their profiles.

## Implementation Details
<!-- Explain how the feature was implemented -->
- Role-based Logic:
Backend get_queryset filters tasks dynamically based on user role and garden membership.
Mobile frontend logic ensures managers see all garden tasks; workers only see what they are assigned or general unassigned tasks.
- Calendar Integration:
Integrated react-native-calendars to visualize tasks.
Marked dates are dynamically colored by task status.
- Task Management:
Added custom backend actions: accept_task, decline_task, complete_task, assign_task.
- Community Logic:
Fetched users with shared garden membership.
Implemented follow and unfollow logic with no approval flow.
Rendered dynamic Follow button only for users not already followed.
- Profile Views:
Implemented /user/[id] route to allow public user profile access.
Reused existing profile structure while hiding sensitive/self-only actions like logout.

## Component
<!-- Mark the appropriate component(s) with an [x] -->
- [ ] Frontend
- [ ] Backend
- [x] Mobile
- [ ] Other (please specify):

## Related Issue(s)
<!-- Link related issues using # syntax -->
<!-- Example: Closes #123, Implements #456 -->
#173

## Screenshots/Videos
<!-- Add screenshots or videos that demonstrate the feature -->
<img width="332" alt="Screenshot 2025-05-12 at 23 57 29" src="https://github.com/user-attachments/assets/f6386659-ea82-4525-9ffb-4059a8990c19" />
<img width="331" alt="mygardens" src="https://github.com/user-attachments/assets/7911c7bc-410c-445f-b557-99df6fec1fc2" />
<img width="347" alt="follow" src="https://github.com/user-attachments/assets/5c75d920-028e-4f84-9992-a1e609cb0329" />
<img width="331" alt="userscreen" src="https://github.com/user-attachments/assets/4a958e68-2e69-4c0a-a67b-b1ffc64d04ef" />
<img width="338" alt="tasksinagarden" src="https://github.com/user-attachments/assets/5071656d-ee40-4870-96eb-d3da52d3cfef" />
<img width="330" alt="Screenshot 2025-05-12 at 23 57 11" src="https://github.com/user-attachments/assets/fbabaea9-ef30-4c6a-8984-b6b1d283e457" />
<img width="336" alt="Screenshot 2025-05-12 at 23 56 46" src="https://github.com/user-attachments/assets/2bb2caa0-8ac9-4d50-a84d-24720e9b0f20" />
<img width="325" alt="calendarscreen" src="https://github.com/user-attachments/assets/d58144da-dfa5-486c-938c-0b3c922c050c" />
<img width="348" alt="Screenshot 2025-05-12 at 23 58 11" src="https://github.com/user-attachments/assets/d27d8884-7fea-4993-bef9-d0b9fa0203ef" />
<img width="348" alt="Screenshot 2025-05-12 at 23 58 20" src="https://github.com/user-attachments/assets/ca2bc1c1-5a54-4f41-b255-3b7245ea6d48" />
<img width="354" alt="Screenshot 2025-05-12 at 23 58 44" src="https://github.com/user-attachments/assets/a199f606-3e74-4c1e-82bf-bddab1101134" />
<img width="329" alt="Screenshot 2025-05-12 at 23 59 47" src="https://github.com/user-attachments/assets/78aaca4f-51e1-4dea-85de-06979c799397" />
<img width="349" alt="Screenshot 2025-05-13 at 00 00 07" src="https://github.com/user-attachments/assets/9104ae37-4ab0-465c-842b-2d02642640a6" />
<img width="331" alt="Screenshot 2025-05-13 at 00 00 35" src="https://github.com/user-attachments/assets/aab2f4d7-8acb-4af5-a34d-ccb5fd4ffda4" />
<img width="324" alt="Screenshot 2025-05-13 at 00 00 42" src="https://github.com/user-attachments/assets/0fc430ec-1b72-4378-91b2-20a322a8e4de" />
<img width="347" alt="Screenshot 2025-05-13 at 00 01 37" src="https://github.com/user-attachments/assets/6df7a6e8-ea72-4a74-a045-1ef141470638" />
<img width="339" alt="muinprogresstask" src="https://github.com/user-attachments/assets/5b2d59dd-82ec-47d5-a8b0-707e9acda601" />
<img width="321" alt="mytasks" src="https://github.com/user-attachments/assets/442e56e9-028a-4474-954c-9071f4710d0a" />
<img width="342" alt="Screenshot 2025-05-13 at 00 01 42" src="https://github.com/user-attachments/assets/299c755e-fd4d-4438-a5e5-a5c1bf671cdb" />
<img width="341" alt="Screenshot 2025-05-13 at 00 23 17" src="https://github.com/user-attachments/assets/1f5f30b9-9b34-4ed2-b012-89bab510dca6" />




## Testing Strategy
<!-- Describe how you tested the feature -->
- [ ] Unit tests
- [ ] Integration tests
- [x] Manual testing
- [ ] Other (please specify):

## Notes for Reviewers
<!-- Anything specific that reviewers should focus on or know about -->
-
